<?xml version="1.0"?>
<robot name="arm" xmlns:xacro="http://ros.org/wiki/xacro">


  <!-- Material and Colors-->
  <material name="black">
    <color rgba="${0/255} ${0/255} ${0/255} 1"/>
  </material>
  <material name="dark_grey">
    <color rgba="${105/255} ${105/255} ${105/255} 1"/>
  </material>
  <material name="grey">
    <color rgba="${128/255} ${128/255} ${128/255} 1"/>
  </material>
  <material name="light_grey">
    <color rgba="${221/255} ${221/255} ${221/255} 1"/>
  </material>
  <material name="white">
    <color rgba="${255/255} ${255/255} ${255/255} 1"/>
  </material>
  <material name="red">
    <color rgba="${255/255} ${0/255} ${0/255} 1"/>
  </material>


    <!-- General Variables-->
    <xacro:property name="pi" value="3.1415269" />

    <!-- Mesh files of exact geometry-->
    <xacro:property name="mesh_file_baseplate" value="package://mav_arm_force_controlled/meshes/geometry/baseplate.stl" />
    <xacro:property name="mesh_file_dyamixel" value="package://mav_arm_force_controlled/meshes/geometry/dynamixel.stl" />
    <xacro:property name="mesh_file_connecting" value="package://mav_arm_force_controlled/meshes/geometry/connecting.stl" />
    <xacro:property name="mesh_file_hebi" value="package://mav_arm_force_controlled/meshes/geometry/hebi.stl" />
    <xacro:property name="mesh_file_hebitotube2" value="package://mav_arm_force_controlled/meshes/geometry/hebitotube2.stl" />
    <xacro:property name="mesh_file_tube2" value="package://mav_arm_force_controlled/meshes/geometry/tube2.stl" />
    <xacro:property name="mesh_file_clamp2" value="package://mav_arm_force_controlled/meshes/geometry/tube2todynamixel.stl" />
    <xacro:property name="mesh_file_actuator1" value="package://mav_arm_force_controlled/meshes/geometry/dynamixel.stl" />
    <xacro:property name="mesh_file_clamp1" value="package://mav_arm_force_controlled/meshes/geometry/dynamixeltotube1.stl" />
    <xacro:property name="mesh_file_tube1" value="package://mav_arm_force_controlled/meshes/geometry/tube1.stl" />
    <xacro:property name="mesh_file_tip" value="package://mav_arm_force_controlled/meshes/geometry/tip.stl" />

    <!-- Mesh files of simplified geometry for collision checking-->
    <xacro:property name="s_mesh_file_baseplate" value="package://mav_arm_force_controlled/meshes/simplified_geometry/baseplate.stl" />
    <xacro:property name="s_mesh_file_dyamixel" value="package://mav_arm_force_controlled/meshes/simplified_geometry/dynamixel.stl" />
    <xacro:property name="s_mesh_file_connecting" value="package://mav_arm_force_controlled/meshes/simplified_geometry/connecting.stl" />
    <xacro:property name="s_mesh_file_hebi" value="package://mav_arm_force_controlled/meshes/simplified_geometry/hebi.stl" />
    <xacro:property name="s_mesh_file_hebitotube2" value="package://mav_arm_force_controlled/meshes/simplified_geometry/hebitotube2.stl" />
    <xacro:property name="s_mesh_file_tube2" value="package://mav_arm_force_controlled/meshes/simplified_geometry/tube2.stl" />
    <xacro:property name="s_mesh_file_clamp2" value="package://mav_arm_force_controlled/meshes/simplified_geometry/tube2todynamixel.stl" />
    <xacro:property name="s_mesh_file_actuator1" value="package://mav_arm_force_controlled/meshes/simplified_geometry/dynamixel.stl" />
    <xacro:property name="s_mesh_file_clamp1" value="package://mav_arm_force_controlled/meshes/simplified_geometry/dynamixeltotube1.stl" />
    <xacro:property name="s_mesh_file_tube1" value="package://mav_arm_force_controlled/meshes/simplified_geometry/tube1.stl" />
    <xacro:property name="s_mesh_file_tip" value="package://mav_arm_force_controlled/meshes/simplified_geometry/tip.stl" />


    <!-- Mesh files are in mm. All mesh files are scaled by the factor 1000-->
    <xacro:property name="mesh_scale" value="0.001" />


    <!-- The Mass and inertia elements are defined before the link element-->

    <xacro:property name="baseplate_mass" value="0.040063354" />


    <xacro:property name="baseplate_cog_x" value="0.025202079" />
    <xacro:property name="baseplate_cog_y" value="0" />
    <xacro:property name="baseplate_cog_z" value="0.0025" />

    <xacro:property name="baseplate_ixx" value="0.000063973" />
    <xacro:property name="baseplate_ixy" value="0" />
    <xacro:property name="baseplate_ixz" value="0.000005877" />
    <xacro:property name="baseplate_iyy" value="0.000151767" />
    <xacro:property name="baseplate_iyz" value="0" />
    <xacro:property name="baseplate_izz" value="0.000214186" />

  <link name="world"/>

  <joint name="fixed" type="fixed">
    <parent link="world"/>
    <child link="baseplate"/>
    <origin xyz="0.5 0.5 0" rpy="0 0 0" />
  </joint>

  <link name="baseplate">
    <visual>
      <geometry>
          <mesh filename="${mesh_file_baseplate}"
                scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
       <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_baseplate}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${baseplate_mass}"/>
      <inertia ixx="${baseplate_ixx}" ixy="${baseplate_ixy}" ixz="${baseplate_ixz}" iyy="${baseplate_iyy}" iyz="${baseplate_iyz}" izz="${baseplate_izz}" />
    </inertial>
  </link>

  <joint name="baseplate_to_actuator3" type="fixed">
    <parent link="baseplate"/>
    <child link="actuator3"/>
    <origin rpy="0 0 ${-pi/2}" xyz="${0.0425+0.0055} 0 ${.005+0.017}"/>
  </joint>



  <xacro:property name="dynamixel_mass" value="0.126" />

  <xacro:property name="dynamixel_cog_x" value="0.000000008" />
  <xacro:property name="dynamixel_cog_y" value="-0.01519089" />
  <xacro:property name="dynamixel_cog_z" value="0.000362847" />

  <xacro:property name="dynamixel_ixx" value="0.000129338" />
  <xacro:property name="dynamixel_ixy" value="0" />
  <xacro:property name="dynamixel_ixz" value="0" />
  <xacro:property name="dynamixel_iyy" value="0.000047045" />
  <xacro:property name="dynamixel_iyz" value="0.0000014" />
  <xacro:property name="dynamixel_izz" value="0.000115625" />


  <link name="actuator3">
    <visual>
      <geometry>
          <mesh filename="${mesh_file_dyamixel}"
                scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_dyamixel}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${dynamixel_mass}"/>
      <inertia ixx="${dynamixel_ixy}" ixy="${dynamixel_ixy}" ixz="${dynamixel_ixz}" iyy="${dynamixel_iyy}" iyz="${dynamixel_iyz}" izz="${dynamixel_izz}" />
    </inertial>
  </link>


  <joint name="actuator_to_connectingbox" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="actuator3"/>
    <child link="connectingbox"/>
    <origin rpy="0 0 ${pi}" xyz="0 0 ${0.0235+0.0035}"/>
    <limit effort="30" velocity="1.0"/>
  </joint>


  <transmission name="actuator3_dynamixel_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="actuator3_dynamixel_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="actuator_to_connectingbox">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>



  <xacro:property name="connectionmx64hebi_mass" value=" 0.402140403" />

  <xacro:property name="connectionmx64hebi_cog_x" value="0.000000171" />
  <xacro:property name="connectionmx64hebi_cog_y" value="0.000278026" />
  <xacro:property name="connectionmx64hebi_cog_z" value="0.038471524" />

  <xacro:property name="connectionmx64hebi_ixx" value="0.001105121" />
  <xacro:property name="connectionmx64hebi_ixy" value="0.000000016" />
  <xacro:property name="connectionmx64hebi_ixz" value="0" />
  <xacro:property name="connectionmx64hebi_iyy" value="0.001002814" />
  <xacro:property name="connectionmx64hebi_iyz" value="0.000007088" />
  <xacro:property name="connectionmx64hebi_izz" value="0.000170487" />


  <link name="connectingbox">
    <visual>
      <geometry>
        <mesh filename="${mesh_file_connecting}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_connecting}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${connectionmx64hebi_mass}"/>
      <inertia ixx="${connectionmx64hebi_ixx}" ixy="${connectionmx64hebi_ixy}" ixz="${connectionmx64hebi_ixz}" iyy="${connectionmx64hebi_iyy}" iyz="${connectionmx64hebi_iyz}" izz="${connectionmx64hebi_izz}" />
    </inertial>
  </link>




  <joint name="connectingbox_to_actuator2" type="fixed">
    <parent link="connectingbox"/>
    <child link="actuator2"/>
    <origin rpy="${pi/2} ${0} ${pi/2}"  xyz="${0.005+0.015} ${0} ${0.0635}"/>
  </joint>

  <xacro:property name="actuator2_mass" value="0.34" />

  <xacro:property name="actuator2_cog_x" value="0.000000171" />
  <xacro:property name="actuator2_cog_y" value="0.000278026" />
  <xacro:property name="actuator2_cog_z" value="0.038471524" />

  <xacro:property name="actuator2_ixx" value="0.000186786" />
  <xacro:property name="actuator2_ixy" value="0.000041556" />
  <xacro:property name="actuator2_ixz" value="0.000000273" />
  <xacro:property name="actuator2_iyy" value="0.00043422" />
  <xacro:property name="actuator2_iyz" value="0.00000029" />
  <xacro:property name="actuator2_izz" value="0.000554524" />

  <link name="actuator2">
    <visual>
      <geometry>
        <mesh filename="${mesh_file_hebi}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="red"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_hebi}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${actuator2_mass}"/>
      <inertia ixx="${actuator2_ixx}" ixy="${actuator2_ixy}" ixz="${actuator2_ixz}" iyy="${actuator2_iyy}" iyz="${actuator2_iyz}" izz="${actuator2_izz}" />
    </inertial>
  </link>


  <joint name="actuator2_to_clamp3" type="continuous">
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <parent link="actuator2"/>
    <child link="clamp3"/>
    <origin rpy="${pi/2} 0 ${-pi/2}" xyz="${0} 0 ${0.01605}"/>
    <limit effort="30" velocity="1.0"/>
  </joint>


  <transmission name="actuator2_hebi_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="$actuator2_hebi_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="actuator2_to_clamp3">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>


  <xacro:property name="clamp3_mass" value="0.21" />

  <xacro:property name="clamp3_cog_x" value="0.015022968" />
  <xacro:property name="clamp3_cog_y" value="0.001776283" />
  <xacro:property name="clamp3_cog_z" value="0.000017898" />

  <xacro:property name="clamp3_ixx" value="0.000524947" />
  <xacro:property name="clamp3_ixy" value="0.000000017" />
  <xacro:property name="clamp3_ixz" value="-0.00000004" />
  <xacro:property name="clamp3_iyy" value="0.000535726" />
  <xacro:property name="clamp3_iyz" value="-0.000092479" />
  <xacro:property name="clamp3_izz" value="0.000074342" />

  <link name="clamp3">
    <visual>
      <geometry>
        <mesh filename="${mesh_file_hebitotube2}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="black"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_hebitotube2}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${clamp3_mass}"/>
      <inertia ixx="${clamp3_ixx}" ixy="${clamp3_ixy}" ixz="${clamp3_ixz}" iyy="${clamp3_iyy}" iyz="${clamp3_iyz}" izz="${clamp3_izz}" />
    </inertial>
  </link>


  <joint name="clamp3_to_tube2" type="fixed">
    <parent link="clamp3"/>
    <child link="tube2"/>
    <origin rpy="0 ${pi} 0" xyz="${0} ${0.013} ${-0.035}"/>
  </joint>

  <xacro:property name="tube2_mass" value="0.038201767" />

  <xacro:property name="tube2_cog_x" value="0" />
  <xacro:property name="tube2_cog_y" value="0" />
  <xacro:property name="tube2_cog_z" value="0.2" />

  <xacro:property name="tube2_ixx" value="0.002039156" />
  <xacro:property name="tube2_ixy" value="0" />
  <xacro:property name="tube2_ixz" value="0" />
  <xacro:property name="tube2_iyy" value="0.002039156" />
  <xacro:property name="tube2_iyz" value="0" />
  <xacro:property name="tube2_izz" value="0.000003457" />

  <link name="tube2">
    <visual>
      <geometry>
        <mesh filename="${mesh_file_tube2}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="red"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_tube2}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${tube2_mass}"/>
      <inertia ixx="${tube2_ixx}" ixy="${tube2_ixy}" ixz="${tube2_ixz}" iyy="${tube2_iyy}" iyz="${tube2_iyz}" izz="${tube2_izz}" />
    </inertial>
  </link>



    <joint name="tube2_to_clamp2" type="fixed">
      <parent link="tube2"/>
      <child link="clamp2"/>
      <origin rpy="${pi/2} 0 ${-pi}" xyz="${0} ${-0.018} ${0.4-0.055}"/>
    </joint>

    <xacro:property name="clamp2_mass" value="0.128888247" />

    <xacro:property name="clamp2_cog_x" value="0.000000004" />
    <xacro:property name="clamp2_cog_y" value="0.02936423" />
    <xacro:property name="clamp2_cog_z" value="0.016233149" />

    <xacro:property name="clamp2_ixx" value="0.00019462" />
    <xacro:property name="clamp2_ixy" value="0" />
    <xacro:property name="clamp2_ixz" value="0" />
    <xacro:property name="clamp2_iyy" value="0.000067787" />
    <xacro:property name="clamp2_iyz" value="0.000060459" />
    <xacro:property name="clamp2_izz" value="0.000173715" />

    <link name="clamp2">
      <visual>
        <geometry>
          <mesh filename="${mesh_file_clamp2}"
                scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
      </visual>
      <collision>
        <geometry>
          <mesh filename="${s_mesh_file_clamp2}"
                scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
          </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
      </collision>
      <inertial>
        <mass value="${clamp2_mass}"/>
        <inertia ixx="${clamp2_ixx}" ixy="${clamp2_ixy}" ixz="${clamp2_ixz}" iyy="${clamp2_iyy}" iyz="${clamp2_iyz}" izz="${clamp2_izz}" />
      </inertial>
    </link>

      <joint name="clamp2_to_actuator1" type="fixed">
        <parent link="clamp2"/>
        <child link="actuator1"/>
        <origin rpy="0 ${pi} 0" xyz="${0} ${0.0056+0.0425} ${-0.017}"/>
      </joint>

      <xacro:property name="actuator1_mass" value="0.126" />

      <xacro:property name="actuator1_cog_x" value="0.000000008" />
      <xacro:property name="actuator1_cog_y" value="-0.01519089" />
      <xacro:property name="actuator1_cog_z" value="0.000362847" />

      <xacro:property name="actuator1_ixx" value="0.000129338" />
      <xacro:property name="actuator1_ixy" value="0" />
      <xacro:property name="actuator1_ixz" value="0" />
      <xacro:property name="actuator1_iyy" value="0.000047045" />
      <xacro:property name="actuator1_iyz" value="0.0000014" />
      <xacro:property name="actuator1_izz" value="0.000115625" />

    <link name="actuator1">
      <visual>
        <geometry>
            <mesh filename="${mesh_file_actuator1}"
              scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
          </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
      </visual>
      <collision>
        <geometry>
          <mesh filename="${s_mesh_file_actuator1}"
            scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
          </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
      </collision>
      <inertial>
        <mass value="${actuator1_mass}"/>
        <inertia ixx="${actuator1_ixx}" ixy="${actuator1_ixy}" ixz="${actuator1_ixz}" iyy="${actuator1_iyy}" iyz="${actuator1_iyz}" izz="${actuator1_izz}" />
      </inertial>
    </link>

  <joint name="actuator1_to_clamp1" type="continuous">
    <axis rpy="0 0 0" xyz="0 0 1"/>
    <parent link="actuator1"/>
    <child link="clamp1"/>
    <origin rpy="${0} ${0} ${-pi/2}" xyz="${0} ${0} ${0.0235}"/>
    <limit effort="30" velocity="1.0"/>
  </joint>



  <transmission name="actuator1_dynamixel_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <actuator name="$actuator1_dynamixel_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
    <joint name="actuator1_to_clamp1">
      <hardwareInterface>hardware_interface/EffortJointInterface</hardwareInterface>
    </joint>
  </transmission>

  <xacro:property name="clamp1_mass" value="0.11739877" />

  <xacro:property name="clamp1_cog_x" value="0.008717092" />
  <xacro:property name="clamp1_cog_y" value="0.000001711" />
  <xacro:property name="clamp1_cog_z" value="0.011234631" />

  <xacro:property name="clamp1_ixx" value="0.000045773" />
  <xacro:property name="clamp1_ixy" value="-0.00000003" />
  <xacro:property name="clamp1_ixz" value="0.000022588" />
  <xacro:property name="clamp1_iyy" value="0.000079456" />
  <xacro:property name="clamp1_iyz" value="0.000000001" />
  <xacro:property name="clamp1_izz" value="0.00007801" />


  <link name="clamp1">
    <visual>
      <geometry>
        <mesh filename="${mesh_file_clamp1}"
          scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
      </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <material name="red"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="${s_mesh_file_clamp1}"
          scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
      <origin rpy="0 0 0" xyz="0 0 0"/>
    </collision>
    <inertial>
      <mass value="${clamp1_mass}"/>
      <inertia ixx="${clamp1_ixx}" ixy="${clamp1_ixy}" ixz="${clamp1_ixz}" iyy="${clamp1_iyy}" iyz="${clamp1_iyz}" izz="${clamp1_izz}" />
    </inertial>
  </link>

  <joint name="clamp1_to_tube1" type="fixed">
    <parent link="clamp1"/>
    <child link="tube1"/>
    <origin rpy="${0} ${pi/2} ${0}" xyz="${0.015} ${0} ${0.022}"/>
  </joint>

  <xacro:property name="tube1_mass" value="0.02035752" />

  <xacro:property name="tube1_cog_x" value="0" />
  <xacro:property name="tube1_cog_y" value="0" />
  <xacro:property name="tube1_cog_z" value="0.3" />

  <xacro:property name="tube1_ixx" value="0.002443367" />
  <xacro:property name="tube1_ixy" value="0" />
  <xacro:property name="tube1_ixz" value="0" />
  <xacro:property name="tube1_iyy" value="0.002443367" />
  <xacro:property name="tube1_iyz" value="0" />
  <xacro:property name="tube1_izz" value="0.000000929" />

    <link name="tube1">
      <visual>
        <geometry>
          <mesh filename="${mesh_file_tube1}"
            scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="black"/>
      </visual>
      <collision>
        <geometry>
          <mesh filename="${s_mesh_file_tube1}"
            scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
          </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
      </collision>
      <inertial>
        <mass value="${tube1_mass}"/>
        <inertia ixx="${tube1_ixx}" ixy="${tube1_ixy}" ixz="${tube1_ixz}" iyy="${tube1_iyy}" iyz="${tube1_iyz}" izz="${tube1_izz}" />
      </inertial>
    </link>

    <joint name="tube1_to_tip" type="fixed">
      <parent link="tube1"/>
      <child link="tip"/>
      <origin rpy="${0} ${0} ${0}" xyz="${0} ${0} ${0.6}"/>
    </joint>

    <xacro:property name="tip_mass" value="0.039351907" />

    <xacro:property name="tip_cog_x" value="0" />
    <xacro:property name="tip_cog_y" value="0" />
    <xacro:property name="tip_cog_z" value="0.000163952" />

    <xacro:property name="tip_ixx" value="0.000000758" />
    <xacro:property name="tip_ixy" value="0" />
    <xacro:property name="tip_ixz" value="0" />
    <xacro:property name="tip_iyy" value="0.000000758" />
    <xacro:property name="tip_iyz" value="0" />
    <xacro:property name="tip_izz" value="0.0000008" />

    <link name="tip">
      <visual>
      <geometry>
          <mesh filename="${mesh_file_tip}"
            scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
        </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <material name="white"/>
      </visual>
      <collision>
        <geometry>
          <mesh filename="${s_mesh_file_tip}"
            scale="${mesh_scale} ${mesh_scale} ${mesh_scale}"/>
          </geometry>
        <origin rpy="0 0 0" xyz="0 0 0"/>
      </collision>
      <inertial>
        <mass value="${tip_mass}"/>
        <inertia ixx="${tip_ixx}" ixy="${tip_ixy}" ixz="${tip_ixz}" iyy="${tip_iyy}" iyz="${tip_iyz}" izz="${tip_izz}" />
      </inertial>
    </link>


    <gazebo>
      <plugin name="gazebo_ros_control" filename="libgazebo_ros_control.so">
        <robotNamespace>/arm</robotNamespace>
      </plugin>
    </gazebo>

        <!--<robotSimType>gazebo_ros_control/DefaultRobotHWSim</robotSimType>-->

</robot>
